### ISSUE TITLE
DreamArtist : One-Shot Text-to-Image Generation For Diffusers

### ISSUE THREAD

**Issue by adhikjoshi** [huggingface/diffusers#1290]
**Is your feature request related to a problem? Please describe.**
There is new, One-Shot Text-to-Image for Texual Inversion

[Twitter Link](https://twitter.com/nearcyan/status/1591885331837898753)

There is implementation for Automatic1111, can we have same for diffusers?

[Github Code](https://github.com/7eu7d7/DreamArtist-stable-diffusion)

**Describe the solution you'd like**

DreamArtist for Diffusers

**Comment by Pie31415** [comment]
Very interested in this issue. I would like to take this on with some guidance if possible :) .

**Comment by isamu-isozaki** [comment]
@Pie31415 Nice! I might implement it in my fork so you can check that if you want. The main changes for this work seems to be

1. Using negative prompts too to learn what the image is not at the same time as the regular tokens
2. Using a discriminator instead of the noise loss

**Comment by Pie31415** [comment]
@isamu-isozaki Hey, sounds good. Thanks for the first 2 points. Also, do you mind if we work on the issue together? I've just started learning about diffusers so my knowledge is a little limited. Would you possibly be able to give me some guidance or resources to get started?

**Comment by Pie31415** [comment]
Also, this maybe a dumb question, but I'm assuming we're implementing a new pipeline with the 2 previously mentioned changes instead of model?

**Comment by isamu-isozaki** [comment]
@Pie31415 For sure! I guess for resources maybe the textual inversion paper but I was thinking that we can copy the code from the textual inversion, and just make slight changes to do this job. 
We might do our custom pipeline but pipelines are mainly for inference and not training. So I'm thinking first we make the training script-> pipeline.

**Comment by isamu-isozaki** [comment]
I can try setting up some code by the end of the day on my repo

**Comment by Pie31415** [comment]
@isamu-isozaki Thanks a lot for the help. Would you like to move this conversation to discord? I think it'll be more convenient for communication.

**Comment by isamu-isozaki** [comment]
@Pie31415 yup sounds good. My name's Chad Kensington on discord. I'm in the Huggingface discord server.
